src/Cake.Common/Tools/NuGet/SetApiKey/NuGetSetApiKey.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using Cake.Core;
+using Cake.Core.IO;
+using Cake.Core.Utilities;
+
+namespace Cake.Common.Tools.NuGet.SetApiKey
+{
+    /// <summary>
+    /// The NuGet set API key used to set API key used for API/feed authentication
+    /// </summary>
+    public sealed class NuGetSetApiKey : Tool<NuGetSetApiKeySettings>
+    {
+        private readonly ICakeEnvironment _environment;
+        private readonly IToolResolver _nugetToolResolver;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NuGetSetApiKey"/> class.
+        /// </summary>
+        /// <param name="fileSystem">The file system.</param>
+        /// <param name="environment">The environment.</param>
+        /// <param name="processRunner">The process runner.</param>
+        /// <param name="nugetToolResolver">The NuGet tool resolver.</param>
+        public NuGetSetApiKey(IFileSystem fileSystem, ICakeEnvironment environment, 
+            IProcessRunner processRunner, IToolResolver nugetToolResolver)
+            : base(fileSystem, environment, processRunner)
+        {
+            _environment = environment;
+            _nugetToolResolver = nugetToolResolver;
+        }
+
+        /// <summary>
+        /// Installs NuGet packages using the specified package id and settings.
+        /// </summary>
+        /// <param name="apiKey">The API key.</param>
+        /// <param name="source">Server URL where the API key is valid.</param>
+        /// <param name="settings">The settings.</param>
+        public void SetApiKey(string apiKey, string source, NuGetSetApiKeySettings settings)
+        {
+            if (string.IsNullOrWhiteSpace(apiKey))
+            {
+                throw new ArgumentNullException("apiKey");
+            }
+            
+            if (string.IsNullOrWhiteSpace(source))
+            {
+                throw new ArgumentNullException("source");
+            }
+
+            if (settings == null)
+            {
+                throw new ArgumentNullException("settings");
+            }
+
+            Run(settings, GetArguments(apiKey, source, settings), settings.ToolPath);
+        }
+
+        private ProcessArgumentBuilder GetArguments(string apiKey, string source, NuGetSetApiKeySettings settings)
+        {
+            var builder = new ProcessArgumentBuilder();
+
+            builder.Append("setapikey");
+            builder.AppendQuoted(apiKey);
```

Comment:
Shouldn't this be `AppendSecret(AppendQuoted(apiKey))`? Otherwise the output will be shown in the build log.